script.js

```diff
@@ -1,4 +1,569 @@
+  // Loop thru n times, creating a an object with relevant keys each iteration. (n is the number of players in the game)
+  for (var i = 0; i < numPlayers; i += 1) {
+    playerProfiles.push({id: i + 1, diceRolls: [], diceNum: 0, score: 0});
+  }
+};
+
+/**
+ * Return a random number between the specified bounds
+ */
+var getRandNum = function (lowerBound, upperBound) {
+  return Math.floor(Math.random() * upperBound) + lowerBound;
+};
+
+/**
+ * Get hold of a random player in the playerProfiles array, then remove that player
+ */
+var getPlayerRandomly = function () {
+  // generate a random number to select one of the players. Subtract 1 from this number to sync it with array index (becos arrays are 0 indexed)
+  var randomIndex = getRandNum(1, playerProfiles.length) - 1;
+
+  var selectedPlayer = playerProfiles[randomIndex];
+  //Remove player from the array:
+  // -------Method 1--------
+  // // initialise a temp array
+  // var tempArr = [];
+  // // Loop thru the playerProfiles
+  // for (var i = 0; i < playerProfiles.length; i += 1) {
+  //   // If the current index does not match randomIndex, push it into the temp Arr.
+  //   if (i != randomIndex) {
+  //     tempArr.push(playerProfiles[i]);
+  //   }
+  // }
+  // // reassign tempArr to playerProfiles
+  // playerProfiles = tempArr;
+  // -------Method 2--------
+  // Use splice, a JS native method, to remove the element directly from playerProfiles. Info on splice: https://www.w3schools.com/jsref/jsref_splice.asp
+  // In the context we're using splice, we need to specify 2 arguments: 1. The index of the elem you want to remove; 2. The number of elements to remove (starting from the specified index)
+  playerProfiles.splice(randomIndex, 1);
+  // return the selected player
+  return selectedPlayer;
+};
+
+var cloneAnArr = function (anArray) {
+  var copyOfAnArray = [];
+  for (var i = 0; i < anArray.length; i += 1) {
+    copyOfAnArray.push(anArray[i]);
+  }
+  return copyOfAnArray;
+};
+
+var eliminateLoser = function () {
+  // set a variable to hold index of the loser's element.
+  var loserIdx;
+  // loop thru the currRoundParticpants
+  for (var i = 0; i < currRoundParticipants.length; i += 1) {
+    // if an element's id is not the same as the roundWinner's id, that element is the loser's element
+    if (currRoundParticipants[i].id != roundWinner.id) {
+      // record the index of this element
+      loserIdx = i;
+    }
+  }
+
+  // assign the loser's details to a var
+  var loser = currRoundParticipants[loserIdx];
+
+  // remove loser from the currRoundParticipants
+  currRoundParticipants.splice(loserIdx, 1);
+  // return the loser's details
+  return loser;
+};
+
+/*====================================
+==============MAIN====================
+======================================*/
+/**
+ * Play Beat with variable number of players and auto-choosing the largest number from the die rolled.
+ * https://swe101.rocketacademy.co/projects/project-2-dice
```

Comment:
Could we update the URL link to the bootcamp.rocketacademy.co course URL? I believe this one won't work